frontend/src/utilities/useEnableApplication.tsx

```diff
@@ -124,23 +129,18 @@ export const useEnableApplication = (
         enableIntegrationApp(internalRoute, enableValues)
           .then((response) => {
             if (!closed) {
-              if (!response.isInstalled && response.canInstall) {
+              if (
+                response.isInstalled &&
+                response.canInstall &&
+                response.variablesValidationStatus === ''
+              ) {
                 setEnableStatus({ status: EnableApplicationStatus.INPROGRESS, error: '' });
-                return;
-              }
-
-              if (response.isInstalled) {
-                setEnableStatus({
-                  status: EnableApplicationStatus.SUCCESS,
-                  error: response.error,
-                });
-                dispatchResults(undefined);
               }
```

Comment:
By removing this, it looks like you will now have an infinite loader waiting for the key to validate when creating a new key
![image](https://github.com/user-attachments/assets/14506a54-c47b-4133-97b7-87e2424bc3e6)

You can see in the background the "Enable" button went away because it was validated and enabled, but the modal doesn't update

Reply:
@emilys314 The `setEnableStatus({ status: EnableApplicationStatus.INPROGRESS, error: '' });` turn the `status` to `Inprogress`, there is a "watcher" [here](https://github.com/opendatahub-io/odh-dashboard/blob/ca83e34385236beecede593b570de3a5b4279847/frontend/src/utilities/useEnableApplication.tsx#L66) to check the status of the application.